tests/50federation/33room-get-missing-events.pl

```diff
@@ -0,0 +1,140 @@
+test "Outbound federation can request missing events",
+   requires => [ $main::OUTBOUND_CLIENT, $main::INBOUND_SERVER, $main::HOMESERVER_INFO[0],
+                 local_user_and_room_fixtures(),
+                 federation_user_id_fixture() ],
+
+   do => sub {
+      my ( $outbound_client, $inbound_server, $info, $creator, $room_id, $user_id ) = @_;
+      my $first_home_server = $info->server_name;
+
+      my $local_server_name = $inbound_server->server_name;
+      my $datastore         = $inbound_server->datastore;
+
+      my $missing_event;
+
+      $outbound_client->join_room(
+         server_name => $first_home_server,
+         room_id     => $room_id,
+         user_id     => $user_id,
+      )->then( sub {
+         my ( $room ) = @_;
+
+         # Generate but don't send an event
+         $missing_event = $room->create_event(
+            type => "m.room.message",
+
+            sender  => $user_id,
+            content => {
+               body => "Message 1",
+            },
+         );
+
+         # Generate another one and do send it so it will refer to the
```

Comment:
bleh, I don't like that I have no reason to believe that this will add a created-but-unsent event to the prev_events of this event. Could this be made explicit, perhaps by taking an additional_prev_events kwarg, and explicitly passing $missing_event->event_id to it?